From dd15cc48155c96598be3190a9a2a389bcef169b9 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Fri, 18 Oct 2013 10:26:43 -0700
Subject: [PATCH 304/382] MIPS: OCTEON: Synchronize CP0.Count across CPUs.

Based on SDK octeon3_3.10.

Needed for KVM to work with cn70XX-P1

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/cavium-octeon/csrc-octeon.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/arch/mips/cavium-octeon/csrc-octeon.c b/arch/mips/cavium-octeon/csrc-octeon.c
index ffe128a..05493a2 100644
--- a/arch/mips/cavium-octeon/csrc-octeon.c
+++ b/arch/mips/cavium-octeon/csrc-octeon.c
@@ -91,6 +91,12 @@ void octeon_init_cvmcount(void)
 			}
 		}
 		write_c0_cvmcount(clk_count);
+		/*
+		 * cn70XX-P1 core-19049 requires synchronized Count for
+		 * KVM guest timers to report with required GTOffset
+		 * of 0.
+		 */
+		write_c0_count((u32)clk_count);
 	}
 	local_irq_restore(flags);
 }
-- 
1.7.0.4

